borgmatic/config/schema.yaml

```diff
@@ -1602,6 +1602,88 @@ properties:
                 example:
                     - start
                     - finish
+    zabbix:
+        type: object
+        additionalProperties: false
+        properties:
+            itemid:
+                type: integer
+                description: |
+                    The itemid to publish to
+                    for details.
+                example: 55105
+            host:
+                type: string
+                description: |
+                    Host name where the item is stored.
+                    Required if 'itemid' is not set
+                example: borg-server
+            key:
+                type: string
+                description: |
+                    Key of the host where the item is stored.
+                    Required if 'itemid' is not set
+                example: borg.status
+            server:
+                type: string
+                description: |
+                    The address of your self-hosted zabbix instance.
+                example: https://zabbix.your-domain.com
+            username:
+                type: string
+                description: |
+                    The username used for authentication.
+                example: testuser
+            password:
+                type: string
+                description: |
+                    The password used for authentication.
+                example: fakepassword
+            api_key:
+                type: string
+                description: |
+                    The api_key used for authentication.
+                example: fakekey
+            start:
+                type: object
+                properties:
+                    value:
```

Comment:
Depending on how the Zabbix admin has configured the item, it may store data as a string, integer etc. Being able to set the value will let you sent it to whatever you want. Hopefully that answered your question 😃

Reply:
Not exactly.. unless I'm misunderstanding something. I was just asking why `value:` is a separate sub-option under `start:`, `finish:`, etc. Rather than, say, making `start:` or `finish:` just have a string/scalar value and no sub-options.

It's not really important.. I was just wondering the rationale, since there's only a single sub-option right now. But I think the way you have it is fine.

Reply:
Ah! I thought it needed to be a sub option.

The more I think of it though, value should be allowed to be either a string, or an integer. Is there a way for that to happen?

It seems like things are statically set in the schema

Reply:
I don't think it absolutely needs to be a sub option, but it's totally fine if it is. At minimum it's more consistent with the other monitoring hooks that way.

Anyway, if you'd like the value to be either a string or an integer, I can think of two different ways:

1. I think [`type:`'s value can be a list](https://cswr.github.io/JsonSchema/spec/multiple_types/). So for instance: `type: [string, integer]`. Or if you prefer:

```yaml
type:
    - string
    - integer
```

2. If that doesn't work or if you'd like a different approach, you can use [`oneOf`](https://opis.io/json-schema/2.x/multiple-subschemas.html) to declare multiple different sub-schemas—one with a string option and one with an integer option. This is probably more verbose than the previous approach though. `oneOf` is already used elsewhere in the schema, so you can get an idea of how it's applied.

Of course with either of these approaches, your code in `zabbix.py` to consume the value will have to take into account that it can be one of two different types.

Hope that helps!

Reply:
fixed here: https://github.com/borgmatic-collective/borgmatic/commit/60c5949c23cc6eb66f88d6d4d9b118ee43fde78e